src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -316,9 +322,23 @@ public DatasetSummaryModel patch(
     if (!patchSucceeded) {
       throw new RuntimeException("Dataset was not updated");
     }
+    // TODO - what should we do if they're turning off secure monitoring?
+    if (patchRequest.isSecureMonitoringEnabled()) {
+      enableSecureMonitoring(id);
```

Comment:
Is there any harm to running this method if the dataset already had secure monitoring enabled?

Reply:
Very good question -- it's very possible but not certain that the refoldering request could fail.

Reply:
Cool.  A failure isn't necessarily a bad thing here, as long as it doesn't leave things in a bad state.  We don't have access to resource locking as part of the patch endpoint which is how we might handle this in a flight (not arguing for a flight, I think it might be overkill given the context of this request).

Reply:
I added a bit of logic to skip the project if it is already in the secure monitoring folder. I've also added to a test to confirm that this works okay!

Reply:
Awesome!